Without this patch compiling asio as a separate library will fail on Mingw with
undefined reference to vtable for the `bad_executor` exception.

--- a/boost/asio/execution/bad_executor.hpp
--- b/boost/asio/execution/bad_executor.hpp
27c27
< class bad_executor
---
> class BOOST_ASIO_DECL bad_executor
32c32,35
<   BOOST_ASIO_DECL bad_executor() noexcept;
---
>   bad_executor() noexcept;
> 
>   /// Destructor.
>   virtual ~bad_executor() noexcept;
35c38
<   BOOST_ASIO_DECL virtual const char* what() const noexcept;
---
>   virtual const char* what() const noexcept;

--- a/boost/asio/execution/impl/bad_executor.ipp
+++ b/boost/asio/execution/impl/bad_executor.ipp
30a31,34
> bad_executor::~bad_executor() noexcept
> {
> }
> 
